pkg/docker/docker.go

```diff
@@ -256,35 +273,58 @@ type BuildImageOptions struct {
 }
 
 // New creates a new implementation of the STI Docker interface
-func New(config *api.DockerConfig, auth docker.AuthConfiguration) (Docker, error) {
-	var client *docker.Client
-	var err error
+func New(config *api.DockerConfig, auth api.AuthConfig) (Docker, error) {
+	var client *dockerapi.Client
+	var httpClient *http.Client
 	if config.CertFile != "" && config.KeyFile != "" && config.CAFile != "" {
-		client, err = docker.NewTLSClient(
-			config.Endpoint,
-			config.CertFile,
-			config.KeyFile,
-			config.CAFile)
-	} else {
-		client, err = docker.NewClient(config.Endpoint)
+		tlscOptions := tlsconfig.Options{
+			CAFile:   config.CAFile,
+			CertFile: config.CertFile,
+			KeyFile:  config.KeyFile,
+		}
+		tlsc, tlsErr := tlsconfig.Client(tlscOptions)
+		if tlsErr != nil {
+			return nil, tlsErr
+		}
+		httpClient = &http.Client{
+			Transport: k8snet.SetTransportDefaults(&http.Transport{
+				TLSClientConfig: tlsc,
+			}),
+		}
 	}
+
+	client, err := dockerapi.NewClient(config.Endpoint, "", httpClient, nil)
```

Comment:
@csrwng is correct; in fact, when I put a dummy/empty httpClient when no cert/key/CA file, that caused problems.